modules/tests/src/test/scala/dev/profunktor/redis4cats/TestScenarios.scala

```diff
@@ -575,4 +578,49 @@ trait TestScenarios { self: FunSuite =>
       _ <- IO(assert(c4 > 0, "merged hyperloglog should think it has more than 0 items in"))
     } yield ()
   }
+
+  def keyPatternSubScenario(client: RedisClient): IO[Unit] = {
+    import dev.profunktor.redis4cats.effect.Log.NoOp._
+    import scala.language.postfixOps
+
+    val pattern = "__keyevent*__:*"
+    val key     = "somekey"
+    val resources = for {
+      ref <- Resource.eval(Ref.of[IO, Option[RedisPatternEvent[String, String]]](None))
+      commands <- Redis[IO].fromClient(client, RedisCodec.Utf8)
+      sub <- PubSub.mkSubscriberConnection[IO, String, String](client, RedisCodec.Utf8)
+      stream <- Resource.pure(sub.psubscribe(RedisPattern(pattern)))
```

Comment:
Should this be `eval` instead of `pure`?

Reply:
No, it should be `pure`. It returns a `fs2.Stream` instead of an `IO` :-)